src/ucs/datastruct/string_buffer.c

```diff
@@ -174,3 +174,16 @@ void ucs_string_buffer_dump(const ucs_string_buffer_t *strb,
         fputs(tok, stream);
     }
 }
+
+char *ucs_string_buffer_extract_mem(const ucs_string_buffer_t *strb)
+{
+    char *c_str;
+
+    if (ucs_array_is_fixed(&strb->str)) {
+        c_str = ucs_strdup(ucs_array_begin(&strb->str), "ucs_string_buffer");
+    } else {
+        c_str = ucs_array_begin(&strb->str);
+    }
+
+    return c_str;
```

Comment:
reset string buffer internal pointer to NULL and length/capacity to 0

Reply:
Added code to reset the array fields. But for dynamic only to make it similar to existing flow when  `ucs_string_buffer_cleanup` is called only  when `ucs_string_buffer_init` is used.